pi-manage

```diff
@@ -1286,8 +1289,130 @@ def import_tokens(file, tokenrealm=None):
 token_manager.add_command('import', Command(import_tokens))
 
 
-# conf export menu
+from functools import wraps, partial
+import inspect
+import json
+
+fmt_dict = {'python': str,
+            'json': partial(json.dumps, indent=2),
+            'yaml': yaml.safe_dump}
+
+
+def exporter_command(func):
+    """ Return a flask_script Command object for exporting data"""
+
+    opts = []
+    # get parameter of function to call
+    arguments, varargs, keywords, defaults = inspect.getargspec(func)
+    if 'name' in arguments:
+        opts.append(Option('-n', '--name',
+                           help='Name of the specific object to export'))
+    if 'id' in arguments:
+        opts.append(Option('-i', '--id',
+                           help='Id of the specific object to export'))
+    if 'censor' in arguments:
+        opts.append(Option("-c", "--censor", action="store_true",
+                           help="Censor sensitive data from the output"))
+
+    opts.append(Option('-o', '--output',
+                       help='The filename to export the data to,'
+                            ' default is <stdout>'))
+    opts.append(Option('-f', '--format', default='python',
+                       help='output format (*python*, json, yaml)'))
+
+    def exporter_wrapper(f):
+        # Add the output routines to the export function
+        @wraps(f)
+        def exp_wrapper(*args, **kwargs):
+            of = kwargs.pop('output', None)
+            fmt = kwargs.pop('format', 'python')
+            res = f(*args, **kwargs)
+            # format output according to the format parameter
+            if fmt not in fmt_dict:
+                print('Format {0!s} is not defined, using "python" instead.'.format(fmt))
+                fmt = 'python'
+            res = fmt_dict.get(fmt)(res) + '\n'
+            if of and not of == '-':
+                with open(of, 'w') as outfile:
+                    outfile.write(res)
+            else:
+                sys.stdout.write(res)
+        return exp_wrapper
+
+    # now create the command from the wrapper function
+    cmd = Command(exporter_wrapper(func))
+    # and add all the options
+    for o in opts:
+        cmd.add_option(o)
+    return cmd
+
+
+config_exporter_manager = Manager(help='registered exporter commands',
```

Comment:
I think the admin does not care about internal registration, so a help text like "list available exportable sections" might be more straightforward for the admin.

Reply:
Rewrote manager in 426389b